queries/java/locals.scm

```diff
@@ -1,10 +1,37 @@
-; CREDITS @maxbrunsfeld (maxbrunsfeld@gmail.com)
-
+; SCOPES
 (class_declaration
-  name: (identifier) @name) @class
+   body: (_)) @scope
+(method_declaration) @scope
+(block) @scope
+; 3 captures for `if` - if+else, if only, else only
+(if_statement) @scope
+(if_statement
+  body: (_)) @scope
+(if_statement
+  alternative: (_)) @scope
+; 3 captures for`try` - try+catch, try only, catch only
+(try_statement) @scope
+(try_statement
+  body: (_) @scope)
```

Comment:
Why having it twice ?

Reply:
`(try_statement) @scope` captures try+catch
`(try_statement
  body: (_) @scope)`  captures body of try

Reply:
Not that pleased by that then, why not capturing the `try` on one side and `catch` on the other side ?

Reply:
wait

Reply:
yanking the whole exception handling would be nice

so try+catch together

Reply:
But all the body queries are unnecessary. Just use `(block) @scope` and you are done (there's only a few exceptions, e.g. the curly braces of a class are not `block`). You only need to capture the whole thing if it is allowed to introduce own variables like `for` or `catch`.

Reply:
https://imgur.com/z6pjf8Y

Reply:
how about
```scheme
(if_statement) @scope ; deleted "if body" and "else", "if" is needed in additon to "block" for if+else
(try_statement) @scope ; deleted "try body", "try" is needed in additon to "block" for try+catch
(catch_clause) @scope ; "catch" is needed in additon to "block" because it has its own Exception variable

(for_statement) @scope
(while_statement) @scope; deleted "do" but "catch" is needed in additon to "block" because it has its loop variable
```

Reply:
https://imgur.com/z6pjf8Y

Reply:
wait `do` is also needed because it has loop variable in
```java
do{

} while(i-->0);
```

Reply:
That's not a declaration ?

Reply:
What about `do int i; while(true);`? Is this allowed?

Reply:
> What about `do int i; while(true);`? Is this allowed?

yes

Reply:
in this case `(do_statement)` covers the whole thing,
```scheme
(do_statement
  body: (_) @scope)
```
covers the `int i`
![image](https://user-images.githubusercontent.com/36493671/88015476-e3e3f600-cb3e-11ea-91c9-c47938fa43b7.png)

Reply:
which one should i keep?